kvwrapper_etcd/kvwrapper_etcd.go

```diff
@@ -52,7 +67,11 @@ func (e EtcdWrapper) GetVal(key string) (*kvwrapper.KeyValue, error) {
 
 // GetVal returns a []KeyValue found at key
 func (e EtcdWrapper) GetList(key string, sort bool) ([]*kvwrapper.KeyValue, error) {
-	r, err := e.client.Get(key, sort, true)
+	options := &etcd.GetOptions{
+		Sort:      sort,
+		Recursive: true,
+	}
+	r, err := e.kapi.Get(context.Background(), key, options)
 	if err != nil {
 		if strings.HasPrefix(err.Error(), "501:") {
```

Comment:
Curious, y are we interpreting the errors from etcd? Does it matter what error it returns ? As far as we are concerned we got an error. Log it and bubble it up. Tomm if we upgrade  etcd again, and they introduce a bunch of new 5xx error codes, we will break.